### ISSUE TITLE
Build fails because implicit dependency of VerifyReleaseTask

### ISSUE THREAD

**Issue by david0** [allegro/axion-release-plugin#710]
with axion 14.4.3 and Gradle 8.1.1 I get the following error when building:

Reason: Task ':verifyRelease' uses this output of task ':app:compileJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

<details open>
  <summary>Complete output</summary>

```
./gradlew test release                                                            

FAILURE: Build failed with an exception.

* What went wrong:
Some problems were found with the configuration of task ':verifyRelease' (type 'VerifyReleaseTask').
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:compileJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:compileJava' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:compileJava' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:compileJava' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:compileTestJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:compileTestJava' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:compileTestJava' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:compileTestJava' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:test' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:test' as an input of ':verifyRelease'.────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
      2. Declare an explicit dependency on ':app:test' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:test' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:processResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:processResources' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:processResources' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:processResources' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/x/Downloads/axiontest'.
    
    Reason: Task ':verifyRelease' uses this output of task ':app:processTestResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':app:processTestResources' as an input of ':verifyRelease'.
      2. Declare an explicit dependency on ':app:processTestResources' from ':verifyRelease' using Task#dependsOn.
      3. Declare an explicit dependency on ':app:processTestResources' from ':verifyRelease' using Task#mustRunAfter.
    
    Please refer to https://docs.gradle.org/8.1.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```
</details>

That seems to be fixed in 1.15.4, but unfortunately we are stuck with Java 8.

**Comment by david0** [comment]
Sad to hear, so the 1.14.x branch will not receive any more backports?

**Comment by bgalek** [comment]
I don't plan to do so, but If you make a PR i'll can help you with a release

**Comment by david0** [comment]
Fixed in 1.14.5. Thanks!
